apps/brandfolder/src/index.js

```diff
@@ -0,0 +1,87 @@
+import { setup } from 'shared-dam-app';
+
+const CTF_APP_URL = 'https://integration-contentful-app.brandfolder-svc.com'
+const BF_EMBED_URL = `https://integration-panel-ui.brandfolder-svc.com?channel=message&appName=Contentful&origin=${CTF_APP_URL}&initMsg=hi`
+
+
+const CTA = 'Select an asset on Brandfolder';
+
+function makeThumbnail(attachment) {
+  const thumbnail = attachment.thumbnail_url || attachment.url
+  const url = typeof thumbnail === 'string' ? thumbnail : undefined
+  const alt = attachment.filename
+
+  return [url, alt];
+}
+
+
+function renderDialog(sdk) {
+  const config = sdk.parameters.invocation;
+
+  const container = document.createElement('div');
+  container.innerHTML = `<iframe id='brandfolder-embed' class='iframe-container' src='${BF_EMBED_URL}' width=400 height=650 style='border:none;'/>`;
+  document.body.appendChild(container);
+
+  sdk.window.startAutoResizer();
+
+  window.addEventListener('message', e => {
+    const { data, origin } = e
+    if (origin === 'https://integration-panel-ui.brandfolder-svc.com'){
+      const { event, payload } = data;
+      console.log('Brandfolder event', event)
+      if (event == 'selectedAttachment') {
+        sdk.close([payload]);
+      }
+      else if (data.event === 'selectedAsset') {
+        const att_id = payload.attachments[0].id
+        const attachment = payload.included.find(att => att.id === att_id)
+        if (attachment){
+          sdk.close([attachment])
+        }
+      }
+    }
+  });
+
+  // TODO maybe figure out how to make this work
```

Comment:
What were you trying to solve here?

Reply:
Other apps here add the event listener to `document` but I only managed to get it to work on `window`

Reply:
Make sense, we can probably clean up the commented out code then.